core/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -1486,7 +1522,10 @@ protected void alreadyClosed() {
         }
     }
 
-    public void incrementOperationCounter() {
+    public void incrementOperationCounter(long opPrimaryTerm) {
```

Comment:
Can you add a Javadoc for this method that states the primary term is being enforced?